pkg/machine/provider/platform_windows.go

```diff
@@ -40,3 +44,46 @@ func Get() (vmconfigs.VMProvider, error) {
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
 }
+
+// SupportedProviders returns the providers that are supported on the host operating system
+func SupportedProviders() []define.VMType {
+	return []define.VMType{define.HyperVVirt, define.WSLVirt}
+}
+
+// InstalledProviders returns the supported providers that are installed on the host
+func InstalledProviders() ([]define.VMType, error) {
+	installed := []define.VMType{}
+	if wutil.IsWSLInstalled() {
+		installed = append(installed, define.WSLVirt)
+	}
+
+	var outBuf bytes.Buffer
+	cmd := exec.Command("powershell", "Get-WindowsOptionalFeature", "-FeatureName", "Microsoft-Hyper-V-All", "-Online")
```

Comment:
Is there a way to query this without executing powershell?
Output parsing is fragile in my experience given windows likes to prints output in different locales which caused problems for us before.

Reply:
When researching it seemed that  running some sort of command in the terminal and parsing the output was the most common solution. I'll try and find a better way to do this, I didn't think of locale issues coming up

Reply:
Well I am not sure if Get-WindowsOptionalFeature has locale issues, I know we has similar check for WSL that caused did `wsl --version` that caused problems https://github.com/containers/podman/pull/20240

If the powershell command does not has the problems then this might be fine, but I wonder do we not have something like this in libhvee already?

@baude @n1hility can we call something like https://github.com/containers/libhvee/blob/0698042e8c753a95c349a2c45f8a6b04a5aa2e3a/pkg/hypervctl/vmm.go#L28 to know if hyperV is supported or not?

Reply:
i think there is but offhand I don't recall ... @n1hility kind of knew this before  @l0rd might also remember.

Reply:
Installer side I am avoiding parsing the PowerShell command output and I am using the [Win32 API `QueryServiceStatusEx`](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-queryservicestatusex) to query for the status of the `vmcompute` service. But that's in C.

Reply:
Using the libhvee link that @Luap99 sent seems to work